Deprecate `num_steps` in `qml.evolve` (#7954)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

<h3>Deprecations 👋</h3>

* Providing `num_steps` to `qml.evolve` and `Evolution` is deprecated and will be removed in a future version.

>   Instead, you can use :class:`~.TrotterProduct` providing the `n` parameter to perform the

Reviewer: ```suggestion
  Instead, use :class:`~.TrotterProduct` for approximate methods, providing the `n` parameter to perform the
```
